Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/MotionEvents/MotionEventWidget.cpp

```diff
@@ -36,9 +40,13 @@ namespace EMStudio
         layout->addWidget(m_editor, 0, Qt::AlignTop);
     }
 
-
     void MotionEventWidget::ReInit(EMotionFX::Motion* motion, EMotionFX::MotionEvent* motionEvent)
     {
+        m_changeEventHandler.Disconnect();
+        if (motionEvent)
+        {
+            motionEvent->SetEventDataChangeEvent(m_changeEventHandler);
+        }
         m_editor->SetMotionEvent(motion, motionEvent);
```

Comment:
question: Should this be moved inside the null check too?

Reply:
This is the existing logic. It will clear of the data  is null. The api is not clear so we will have to revisit this I think